src/legacy/core_plugins/data/public/filter/filter_bar/filter_editor/lib/filter_editor_utils.ts

```diff
@@ -97,7 +97,7 @@ export function getFilterParams(filter: Filter) {
     case 'range':
       return {
         from: (filter as RangeFilter).meta.params.gte,
-        to: (filter as RangeFilter).meta.params.lt,
+        to: (filter as RangeFilter).meta.params.lte,
```

Comment:
This will fix the filters created by input controls, but it will break filters that were previously created in the filter editor and saved on a dashboard, vis, etc. It looks like I created this bug in the first place when I rewrote the filter bar in React/Typescript. The [original version](https://github.com/elastic/kibana/blob/a3bdf4906797ede139f23bc98e6623fd1bde1aac/src/ui/public/filter_editor/lib/filter_editor_utils.js#L49-L53) of this function accounted for both inclusive and exclusive ranges. Let's recreate that behavior so filters from both sources will work.

Reply:
@nickofthyme it seems I can't do a multi-line suggestion, but I think if you replace lines 99-100 with this it should be equivalent to the old angular version.

```suggestion
        from: (filter as RangeFilter).meta.params.gte || (filter as RangeFilter).meta.params.gt,
        to: (filter as RangeFilter).meta.params.lte || (filter as RangeFilter).meta.params.lt,
```

Reply:
what do you think about this...

```ts
case 'range':
  const { gte, gt, lte, lt } = (filter as RangeFilter).meta.params;
  return {
    from: gte || gt,
    to: lt || lte,
  };
```

that way the current values are accounted for and the values are updated accordingly.

Reply:
That's probably fine, though I'd personally err on the side of retaining the same behavior we used to have in case there is some edge case that relies on it. I don't have a strong opinion on it though.